bpf/lib/common.h

```diff
@@ -350,11 +350,21 @@ struct edt_info {
 
 struct remote_endpoint_info {
 	__u32		sec_identity;
-	__u32		tunnel_endpoint;
+	union {
+		struct {
+			__u32	ip4;
+			__u32	pad1;
+			__u32	pad2;
+			__u32	pad3;
+		};
+		union v6addr	ip6;
+	} tunnel_endpoint;
 	__u16		pad;
 	__u8		key;
 	__u8		flag_skip_tunnel:1,
-			pad2:7;
+			flag_has_tunnel_ep:1,
+			flag_ipv6_tunnel_ep:1,
```

Comment:
Curious, did you consider making this symmetric - or would there be any downside to that approach?

```suggestion
			flag_ipv4_tunnel_ep:1,
			flag_ipv6_tunnel_ep:1,
```

Reply:
Yes, but the common pattern is:
```c
if (do we need to handle tunneling & co.?)
    ... do some stuff ...
    if (is it IPv4 or IPv6 underlay?)
        ... do some more stuff ...
```
So for the first condition, it's easier to have a single flag rather than checking two.

That pattern is maybe not that obvious from the current code because we still have a bunch of places that only support IPv4 anyway (egressgw, VTEP, etc.), but as we go it should become the norm.

Reply:
Agree, in the bigger picture it should behave very similar / we should be able to build the same kind of wrappers around those individual bits to make it look the same.